packages/engine-core/src/data-proxy/DataProxyEngine.ts

```diff
@@ -34,24 +39,29 @@ export class DataProxyEngine extends Engine {
     this.env = this.config.env ?? {}
     this.inlineSchema = config.inlineSchema ?? ''
     this.inlineDatasources = config.inlineDatasources ?? {}
+    this.inlineSchemaHash = config.inlineSchemaHash ?? ''
 
     this.logEmitter = new EventEmitter()
     this.logEmitter.on('error', () => {})
 
-    this.initPromise = this.init()
-  }
-
-  /**
-   * !\ Asynchronous constructor that inits the properties marked with `!`.
-   * So any function that uses such a property needs to await `initPromise`.
-   */
-  private async init() {
-    // we set the network stuff up for the engine to make http calls to the proxy
     const [host, apiKey] = this.extractHostAndApiKey()
-    this.schemaHash = await createSchemaHash(this.inlineSchema)
     this.clientVersion = getClientVersion(this.config)
     this.headers = { Authorization: `Bearer ${apiKey}` }
     this.host = host
+
+    const promise = Promise.resolve() // hack for cloudflare
+    this.pushPromise = promise.then(() => this.pushSchema())
```

Comment:
We perform upfront schema uploads to save time, as soon as the engine instantiates and before any request has been made. This is also an optimization that helped save around  `1s` for when we are running the DP on Vercel Edge.